allocations/static/allocations/js/main.js

```diff
@@ -1 +1,750 @@
+    var e = {};
+    return e.projects = [],
+    e.userProjects = [],
+    e.getAllocations = function() {
+        var b = "There was an error loading allocations.";
+        return d.clearMessages("allocations"),
+        d.addLoading("allocations"),
+        a({
+            method: "GET",
+            url: "/admin/allocations/view/",
+            cache: "true"
+        }).then(function(a) {
+            d.removeLoading("allocations"),
+            e.projects = a.data
+        }, function() {
+            d.addMessage("allocations", b, "danger"),
+            d.removeLoading("allocations")
+        })
+    }
+    ,
+    e.getUserAllocations = function(b) {
+        var c = "There was an error loading allocations.";
+        return d.clearMessages("userAllocations"),
+        d.addLoading("userAllocations"),
+        a({
+            method: "GET",
+            url: "/admin/allocations/user/" + b,
+            cache: "true"
+        }).then(function(a) {
+            d.removeLoading("userAllocations"),
+            "error" === a.data.status.toLowerCase() ? d.addMessage("userAllocations", a.data.msg, "danger") : e.userProjects = a.data.result
+        }, function() {
+            d.addMessage("userAllocations", c, "danger"),
+            d.removeLoading("userAllocations")
+        })
+    }
+    ,
+    e.rejectAllocation = function(b) {
+        var c = "rejectAllocation" + b.id
+          , e = "There was an error rejecting this allocation. Please try again or file a ticket if this seems persistent.";
+        return d.clearMessages(c),
+        d.addLoading(c),
+        a({
+            method: "POST",
+            url: "/admin/allocations/approval/",
+            data: b
+        }).then(function(a) {
+            return d.removeLoading(c),
+            "error" === a.data.status ? (d.addMessage(c, e, "danger"),
+            null) : (d.addMessage(c, "This allocation request is rejected successfully.", "success"),
+            a.data.result)
+        }, function() {
+            d.removeLoading(c),
+            d.addMessage(c, e, "danger")
+        })
+    }
+    ,
+    e.approveAllocation = function(b) {
+        var c = "approveAllocation" + b.id
+          , e = "There was an error approving this allocation. Please try again or file a ticket if this seems persistent.";
+        return d.clearMessages(c),
+        d.addLoading(c),
+        a({
+            method: "POST",
+            url: "/admin/allocations/approval/",
+            data: b
+        }).then(function(a) {
+            return d.removeLoading(c),
+            "error" === a.data.status ? (d.addMessage(c, e, "danger"),
+            null) : (d.addMessage(c, "This allocation request is approved successfully.", "success"),
+            a.data.result)
+        }, function() {
+            d.removeLoading(c),
+            d.addMessage(c, e, "danger")
+        })
+    }
+    ,
+    e.waitingAllocation = function(b) {
```

Comment:
This is the main change to this file, right? Man, this is so painful. I declare a moratorium on more JS updates until we rewrite this in something that has a supported build chain!

Reply:
Can you also include this change in to whatever source file this generated file came from? That way we don't accidentally use it when porting this functionality later.